src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Maui
+{
+    /// <summary>
+    /// Users using Xamarin.Forms lower than version 4.8 should be informed to update to a higher version
+    /// before they run this tool.
+    /// </summary>
+    public class XamarinFormsCheck : IUpgradeReadyCheck
+    {
+        private const string CATEGORY = ".NET MAUI Upgrade";
+        private const string UPGRADE_LINK = "https://aka.ms/upgradeassistant/maui/requirements";
+        private const string XAMARIN_FORMS_MESSAGE = "Support for {0} is limited to Xamarin.Forms version 4.8 or higher. To learn more please read: {1}";
+        private const double MINIMUM_XAMARIN_VERSION = 4.8;
+
+        private readonly ILogger<XamarinFormsCheck> _logger;
+
+        public XamarinFormsCheck(ILogger<XamarinFormsCheck> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public string Id => "Xamarin.Forms minimum version check";
+
+        public string UpgradeMessage => string.Format(CultureInfo.InvariantCulture, XAMARIN_FORMS_MESSAGE, CATEGORY, UPGRADE_LINK);
```

Comment:
There's an `SR.Format` method that could be used here to ensure everything is using invariant culture (feel free to add more overloads to that method). That method is shared among all projects.

Reply:
Hmm I saw the other Checks followed this format, so I stuck with it. `SR.Format` is damn cool though, should all the check messages be updated to this format?